x-pack/plugins/ml/public/application/jobs/new_job/pages/components/job_details_step/components/groups/groups_input.tsx

```diff
@@ -62,7 +62,18 @@ export const GroupsInput: FC = () => {
   }
 
   useEffect(() => {
-    setValidation(jobValidator.groupIds);
+    const valid =
+      jobValidator.groupIds.valid === true &&
+      jobValidator.latestValidationResult.groupIdsExist?.valid === true;
+    const message =
+      jobValidator.groupIds.message === undefined
+        ? jobValidator.latestValidationResult.groupIdsExist?.message
+        : jobValidator.groupIds.message;
+
+    setValidation({
+      valid,
+      message,
+    });
```

Comment:
Checked the code of `groups_input.tsx` and it seems like you don't need the combination of `useState` and `useEffect`. You can simply use `useMemo` instead.
```typescript
const validation = useMemo(() => {
// You code from effect here 
return {
  valid,
  message
}
}, [jobValidatorUpdated])
```